### ISSUE TITLE
Старые события в подборках 

### ISSUE THREAD

**Issue by kbochenina** [avatar-project/public-issues#39]
Пользователю предлагаются события, которые уже прошли, рчень плохо смотрится как по мне :(

Есть вариант фильтровать по сегодняшней дате, чтобы попадали только предстоящие?

**Comment by davidvirabian** [comment]
Да, тоже хорошее замечание. Я бы предложил фильтровать только события по сегодняшней дате. Т.к. дата публикации типа выдач "статья" не особо критична.

**Comment by danwhale** [comment]
Мы с @hainamantik пообсуждали, в целом, мы можем сделать разные условия фильтрования прошедших событий, хочется единственного уточнения: 

- некоторым было бы реально интересно посмотреть на те события, коотрые были вчера\позавчера, давайте господа @avatar-project/managers-team попробуем сформировать чёткое и достаточное условие "валидности" события.

**Comment by kbochenina** [comment]
Мое видение по валидности - предстоящие события по отношению к текущей дате-времени.

Здесь сценарий использования ясен: нашел событие - добавил в календрь - аватар напоминает - потом оставил фидбек по событию.

Сценарий использования для прошедших не оч. понятен, тк если он пользуется аватаром по идее эти события ему уже давно показывались, только в режиме "предстоящих". И возникает не оч хорошая коллизия с добавлением в календарь уже прошедших событий.

**Comment by danwhale** [comment]
Можно на уровне фронта сверять даты и деактивировать кнопку.

**Comment by davidvirabian** [comment]
@danwhale Давай так и сделаем

> Можно на уровне фронта сверять даты и деактивировать кнопку. (по вчерашним событиям) а совсем древние мы фильтруем.

**Comment by danwhale** [comment]
> @danwhale Давай так и сделаем
> 
> > Можно на уровне фронта сверять даты и деактивировать кнопку. (по вчерашним событиям) а совсем древние мы фильтруем.

Как сделаем?  Насколько старое событие мы пускаем в подборку? 
@kapc3d  нравится идея?

**Comment by kbochenina** [comment]
Я не поняла, какие аргументы для просмотра прошедших событий. Но лично меня - расстраивает видеть в подборках события, например, с декабря прошлого года. Сделайте 3 суток от текущей даты / времени ЛИБО же те, которые были на текущей неделе.

Это, конечно, в случае если @kapc3d согласится делать деактивацию добавления в календарь на фронте.

Если это сейчас _трудно_, то давайте будем фильтровать ВСЕ прошедшие события. 

Т.е. вам на выбор 2 варианта. Но не должно быть добавления прошедших событий в календарь.

**Comment by kapc3d** [comment]
> Я не поняла, какие аргументы для просмотра прошедших событий. Но лично меня - расстраивает видеть в подборках события, например, с декабря прошлого года. Сделайте 3 суток от текущей даты / времени ЛИБО же те, которые были на текущей неделе.
> 
> Это, конечно, в случае если @kapc3d согласится делать деактивацию добавления в календарь на фронте.
> 
> Если это сейчас _трудно_, то давайте будем фильтровать ВСЕ прошедшие события.
> 
> Т.е. вам на выбор 2 варианта. Но не должно быть добавления прошедших событий в календарь.

не понимаю почему нельзя добавить вчерашнее событие в календарь? человек сходил на какой то ивент просто друзья позвали, а тут ему оно вываливает в подборках, он такой - опа, а я ж ходил, и добавил к себе в календарь.
В общем мне бы не хотелось городить лишние проверки с нашей стороны и ограничивать пользователя только исходя из наших разработческих предположений

**Comment by kbochenina** [comment]
Если это событие было ему интересно, он бы его добавил пока оно в
предстоящих. Ок - деактивации значит сейчас не будет.

Тем не менее, реквестирую фильтр по "старым" событиям со стороны Бэка. Выше
2 варианта приводила.

пт, 19 февр. 2021 г., 20:27 Andrey <notifications@github.com>:

**Comment by hainamantik** [comment]
Мне как пользователю было бы интересно посмотреть на события до ~месяца назад, может я пропустил какие-то конференции, а сейчас смогу глянуть видео с них или другие полезные материалы

**Comment by kbochenina** [comment]
Я в приложении сейчас никаких  конференций с которых можно увидеть что-то полезное, не вижу, зато вижу вот такое:

![photo_2021-02-20_11-36-51](https://user-images.githubusercontent.com/5405987/108589569-f8f06400-736f-11eb-91de-93114dcb00bb.jpg)

Мы можем сделать фильтр для старых событий только по событиям ИТМО? @danwhale @hainamantik @shpaker

**Comment by shpaker** [comment]
Мне кажется, что это надо фильтровать на стороне assets-events. @hainamantik но ведь на корфид это не повлияет? Он же сам все компонует?

**Comment by hainamantik** [comment]
"Старые события в подборках" это относится только к подборкам, или в ленте рекомендаций тоже старьё не должно всплывать?

**Comment by shpaker** [comment]
@avatar-project/backend-team надо впиливать фикс в https://github.com/avatar-project/core-feed чтоб скипался старый и не актуальный контент

**Comment by shpaker** [comment]
> "Старые события в подборках" это относится только к подборкам, или в ленте рекомендаций тоже старьё не должно всплывать?

Как все просто стало прям не могу. Ну кроме того, что мы нагородили кучу уродливых АПИ.

"и я идеей со сторизами пытался это как то упростить" (ц)

**Comment by hainamantik** [comment]
> @avatar-project/backend-team надо впиливать фикс в https://github.com/avatar-project/core-feed чтоб скипался старый и не актуальный контент

вот я и жду указаний, какой контент считать старым)

**Comment by kbochenina** [comment]
> "Старые события в подборках" это относится только к подборкам, или в ленте рекомендаций тоже старьё не должно всплывать?

по идее нигде не должно всплывать


> вот я и жду указаний, какой контент считать старым)

 Сделайте 1 неделю назад от текущей даты / времени (вариант 10 ЛИБО же оставлять только те, которые были с начала текущей неделе (вариант 2). Что проще. Если одинаково, то вариант 1

**Comment by davidvirabian** [comment]
@hainamantik предлагаю считать события старыми, если дата публикации не совпадает с датой, когда этот контент подгружается в подборку.
Это сложно реализовать?

**Comment by shpaker** [comment]
Вот вы сейчас пишете вещи, которые кардинально расходится с тем, что есть у нас на бэкенде и то что мы писали полгода.
> по идее нигде не должно всплывать

Но ведь актуальность статей != актуальность события. Ко всем типам контента такую логику достаточно старнно применять. Я правда и так не очень понимаю как у нас работает алгоритмы компоновки. 

> дата публикации не совпадает с датой, когда этот контент подгружается в подборку.

какая-то невероятная фраза - я её не понял, а то что я понял не хотел бы даже читать (см. ответ выше для Клавдии).

**Comment by shpaker** [comment]
То есть я к тому, что надо актуальность контента определять исходя из его типа и это должен достаточно умный алгоритм.

**Comment by kbochenina** [comment]
@shpaker  я имела в виду чисто для событий! т.е. если в подборках фильтруем старые события, то и в ленте было бы неплохо. Я думала вопрос @hainamantik про это. И весь тикет про события.

Фильтр для событий предложен такой:

Сделайте 1 неделю назад от текущей даты / времени (вариант 1) ЛИБО же оставлять только те, которые были с начала текущей недели (вариант 2). Что проще. Если одинаково, то вариант 1. Имеется в виду по дате события.

То есть если у нас сейчас 20 февраля, то при варианте 1 остаются события начиная с 13 февраля, при варианте 2 - начиная с 15 февраля.
